server/controller/comments/getAllCommentForAPost.go

```diff
@@ -0,0 +1,45 @@
+package comments
+
+import (
+	"campusburn-backend/dbConnection"
+	"campusburn-backend/model"
+
+	"github.com/gofiber/fiber/v2"
+)
+
+/*
+FLOW TO GET ALL THE COMMENTS FOR A POST:
+---------------------------------------
+1. Get the post Id from the request.
+2. Find all the comments for this post from the database.
+3. If any error while finding the comments, then return error response.
+4. Return the success response with all comments as the data.
+
+*/
+
+func GetAllCommentForAPost(c *fiber.Ctx) error {
+
+	var post []model.Post
+	var postId = c.Params("postId")
+
+	postSearchError := dbConnection.DB.Where("ID = ?", postId).First(&post).Error
+	if postSearchError != nil {
+		return c.Status(fiber.StatusBadRequest).JSON(fiber.Map{
+			"Error": "Post not found",
+		})
+	}
+
+	var allComments []string
+
+	commentSearchError := dbConnection.DB.Model(&model.Comment{}).Where("post_id = ?", postId).Pluck("comment_content", &allComments).Error
```

Comment:
Please use `Preload` to fetch the details of the user and post corresponding to this comment. You can ChatGPT about this term, and can let me know if you face any error.

Reply:
surely, I will implement using `Preload` and will walk through `Preload`.
I will let you know about this